### DISCUSSION TITLE
Packaging for MAUI on iOS possible?

### DISCUSSION THREAD

**Discussion by markdav-is** [oqtane/oqtane.framework#3747] [Q&A]
In the image below, you can see my Oqtane client running in iOS and it's missing resources.  Style and JS libraries are loading fine, but what's missing is the client DLLs.  I'm New to MAUI, but have built Xamarin apps.  I'm using the "Link to Mac" feature in visual studio.  Somewhere in the build pipeline, I need to copy my dependent DLLs.  I know I'm taking a risk by attempting an IOS build.  I'm courious if anyone has attempted this.

![image](https://github.com/oqtane/oqtane.framework/assets/311063/fc19e0f0-acc0-41b6-ad5b-684f7f35422d)

The next things I'm going to try:
- research the MAUI targets and how that file is generated.  I see a large comment section that might be useful
- add project references to my client project to  the MAUI just to see if they get picked up as part of the build.

**Comment by leigh-pointer** [comment]
Are they listed in your moduleInfo dependencies?
________________________________
From: Mark Davis ***@***.***>
Sent: Friday, February 9, 2024 7:32:24 AM
To: oqtane/oqtane.framework ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [oqtane/oqtane.framework] Packaging for MAUI on iOS possible? (Discussion #3747)


In the image below, you can see my Oqtane client running in iOS and it's missing resources. Style and JS libraries are loading fine, but what's missing is the client DLLs. I'm New to MAUI, but have built Xamarin apps. I'm using the "Link to Mac" feature in visual studio. Somewhere in the build pipeline, I need to copy my dependent DLLs. I know I'm taking a risk by attempting an IOS build. I'm courious if anyone has attempted this.

image.png (view on web)<https://github.com/oqtane/oqtane.framework/assets/311063/fc19e0f0-acc0-41b6-ad5b-684f7f35422d>

The next things I'm going to try:

  *   research the MAUI targets and how that file is generated. I see a large comment section that might be useful
  *   add project references to my client project to the MAUI just to see if they get picked up as part of the build.

**Reply by markdav-is** [reply]
great tip.  I have not used that before.  I bet that's it.

**Reply by markdav-is** [reply]
it's the module itself that is not found.  are there site dependencies?  I'll dig deeper into the dynamic module loading bit to see how it's resolving types for MAUI.

**Reply by markdav-is** [reply]
I forgot to config the remote site for Hybrid!  Once I did that, the assemblies came down!
